tests/net/socket/misc/src/main.c

```diff
@@ -286,6 +286,8 @@ void test_so_bindtodevice(int sock_c, int sock_s, struct sockaddr *peer_addr,
 	zassert_equal(ret, 0, "close failed, %d", errno);
 	ret = close(sock_s);
 	zassert_equal(ret, 0, "close failed, %d", errno);
+
+	k_sleep(K_MSEC(CONFIG_NET_TCP_TIME_WAIT_DELAY));
```

Comment:
The changes in the unrelated tests are necessary, as otherwise these tests now fail without the REUSEADDR/REUSEPORT option. Most of the tests create sockets in the `TIME_WAIT` state, resulting in the next test failing on a call to `bind()` to the same IP address / port combination.